server/src/main/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverAction.java

```diff
@@ -334,6 +345,40 @@ protected void masterOperation(
         );
     }
 
+    private void initializeFailureStore(
+        RolloverRequest rolloverRequest,
+        ActionListener<RolloverResponse> listener,
+        String trialSourceIndexName,
```

Comment:
Let's see if we manage to have `null` as the source index, if we could drop this argument. I find it a bit confusing that we pass here a source index name when there shouldn't be one.